code/game/objects/structures/aliens.dm

```diff
@@ -82,23 +97,144 @@
 /obj/structure/alien/resin/wall/BlockSuperconductivity()
 	return TRUE
 
-/obj/structure/alien/resin/membrane
-	name = "resin membrane"
-	desc = "Resin just thin enough to let light pass through."
-	icon = 'icons/obj/smooth_structures/alien/resin_membrane.dmi'
-	icon_state = "resin_membrane-0"
-	base_icon_state = "resin_membrane"
-	opacity = FALSE
-	max_integrity = 160
-	resintype = "membrane"
+/*  Resin-Door
+	This borrs a bit of code from Mineral-Door, but is mostly its own thing due to not needing many of the latter's things instead of its own things */
+/obj/structure/alien/resin/door
+	name = "resin door"
+	density = TRUE
+	anchored = TRUE
+	opacity = TRUE
+
+	icon = 'icons/obj/smooth_structures/alien/resin_door.dmi'
+	icon_state = "resin"
+	base_icon_state = "resin"
+	max_integrity = 150
+	armor = list(MELEE = 0, BULLET = 0, LASER = 0, ENERGY = 0, BOMB = 0, BIO = 0, RAD = 0, FIRE = 50, ACID = 50)
+	damage_deflection = 0
+	flags_2 = RAD_PROTECT_CONTENTS_2 | RAD_NO_CONTAMINATE_2
+	rad_insulation = RAD_MEDIUM_INSULATION
+	var/initial_state
+	var/state_open = FALSE
+	var/is_operating = FALSE
+	var/close_delay = 15
+	smoothing_flags = null
+	var/openSound = 'sound/machines/alien_airlock.ogg'
+	var/closeSound = 'sound/machines/alien_airlock.ogg'
```

Comment:
please make these vars follow `snake_case`

Reply:
👍 Fixed here and in Mineral_Doors